modules/ecs-service/scripts/deploy-ecs-service.py

```diff
@@ -0,0 +1,375 @@
+        )
+
+    # The image name can have up to 255 characters, and can only contain alphanumeric characters,
+    # hyphens, underscores, colons, periods, forward slashes, and number signs.
+    if not re.match("^[a-zA-Z0-9-_:./#]+$", args.image) or len(args.image) > 255:
+        raise Exception(
+            "The ECS image name cannot be longer than 255 characters, and can only contain alphanumeric characters, hyphens, underscores, colons, periods, forward slashes, and number signs."
+        )
+
+    # The service name can have up to 255 characters, and can only contain alphanumeric characters,
+    # hyphens, and underscores.
+    if args.service and (
+        not isAlNumHyphenUnderscore(args.service) or len(args.service) > 255
+    ):
+        raise Exception(
+            "The ECS service name cannot be longer than 255 characters and can only contain alphanumeric characters, hyphens, and underscores."
+        )
+
+    # The region must be from the authorized list of regions.
+    if args.region not in ["us-east-1", "us-east-2", "us-west-1", "us-west-2"]:
+        raise Exception(
+            "The region must be one of the following: us-east-1, us-east-2, us-west-1, us-west-2"
+        )
+
+    # Timeout must be greater than 0.
+    if args.timeout <= 0:
+        raise Exception("The timeout must be greater than 0.")
+
+    return args
+
+
+def isAlNumHyphenUnderscore(string) -> bool:
+    """Return True if string is alphanumeric, hyphen, or underscore.
+
+    Args:
+        string (_type_): the string to validate
+
+    Returns:
+        bool: True if string is alphanumeric, hyphen, or underscore. False otherwise.
+    """
+    pattern = "^[a-zA-Z0-9-_]+$"
+    return re.match(pattern, string)
+
+
+def getCurrentTaskDefinition(client: any, cluster: str, service: str) -> (dict, str):
+    """Get the currently deployed task definition in the service.
+
+    Args:
+        client (boto3.ECS.Client): the low level boto3 ECS client
+        cluster (str): the name of the ECS cluster
+        service (str): the name of the ECS service
+
+    Raises:
+        Exception: if the service does not exist in the cluster
+        Exception: if more than one service was found in the cluster with the same name
+
+    Returns:
+        (dict, str): the currently deployed task definition, and image
+    """
+
+    res = client.describe_services(cluster=cluster, services=[service])
+
+    # Assert that the service exists in the cluster
+    if len(res["services"]) == 0:
+        raise Exception("The service does not exist in the cluster")
+    elif len(res["services"]) > 1:
+        raise Exception(
+            "More than one service was found in the cluster with the same name"
+        )
+    logger.info(f"Service {service} found in cluster {cluster}")
+
+    # Log the current task definition
+    current_task_definition = res["services"][0]["taskDefinition"]
+    logger.info(f"Active task definition: {current_task_definition}")
```

Comment:
Nope, because we shouldn't be storing secret values in the task definition. environment variables should be non sensitive and secrets should just be the ARNs of the secrets we are referring to not the actual values.